bsp-testkit/client/TestClient.kt

```diff
@@ -147,6 +151,15 @@ class TestClient(
     }
   }
 
+  fun testCompile(timeout: Duration, params: CompileParams, expectedResult: CompileResult, expectedDiagnostics: List<PublishDiagnosticsParams>) {
+    val transformedParams = applyJsonTransform(params)
+    test(timeout) { session, _ ->
+      val result = session.server.buildTargetCompile(transformedParams).await()
+      assertIterableEquals(expectedDiagnostics, session.client.publishDiagnosticsNotifications)
```

Comment:
maybe either reset the diagnostics state before running the request or change the assertion so it expects *at least* `expectedDiagnostics`

Reply:
Changed to clear at the start.